talk/basicconcepts/auto.tex

```diff
@@ -4,8 +4,8 @@
   \frametitlecpp[11]{Auto keyword}
   \begin{block}{Reason of being}
     \begin{itemize}
-    \item many type declarations are redundant
-    \item and lead to compiler errors if you mess up
+    \item Many type declarations are redundant
+    \item They are also a source for compiler warnings and errors
```

Comment:
You may want to add: unnecessary conversions.

Especially alternating between float/double or int, unsigned, size_t, because that's what you thought the type was, can sometimes create actual conversion code the CPU needs to perform. Or may choose the wrong overload of `std::sin`.

Reply:
Yes, I will add this to the slides if it fits.